### PR TITLE
Add examples of extension libraries.

### PR DESCRIPTION
**Pull request by Revathyvenugopal162**
WIP

### CODE REVIEW

**Comment by jorgepiloto** [comment]
I do not think we are allowed to use those images or distribute those, @Revathyvenugopal162.

Also, you are executing the download of the examples during the CI/CD. That's smart but limits ourselves to only generate examples during the pipeline execution.

In my opinion, we should use a similar logic to the one in [pyansys/actions], where by using [sphinx-jinja], we can inject any sort of content into an RST file.

**Comment by Revathyvenugopal162** [comment]
> I do not think we are allowed to use those images or distribute those, @Revathyvenugopal162.
> 
> Also, you are executing the download of the examples during the CI/CD. That's smart but limits ourselves to only generate examples during the pipeline execution.
> 
> In my opinion, we should use a similar logic to the one in [pyansys/actions], where by using [sphinx-jinja], we can inject any sort of content into an RST file.

I was thinking of adding this code in conf file, also added this image to make sure it is working. I think We can do some modifications in the image.i will look in to using [sphinx-jinja]. But i am open to changes. 
Thanks @jorgepiloto .

File: .github/workflows/ci_cd.yml [hunk 1]
```diff
@@ -45,7 +45,7 @@ jobs:
       - name: "Run Ansys documentation building action"
         uses: pyansys/actions/doc-build@v3
         with:
-          python-version: ${{ env.MAIN_PYTHON_VERSION }}
+          python-version: ${{ env.MAIN_PYTHON_VERSION }}
```
Comment (jorgepiloto) [inline .github/workflows/ci_cd.yml:48]:
```suggestion
          python-version: ${{ env.MAIN_PYTHON_VERSION }}
```

File: .gitignore [hunk 2]
```diff
@@ -28,6 +28,7 @@ share/python-wheels/
 .installed.cfg
 *.egg
 MANIFEST
+sphinx_examples.rst
```
Comment (jorgepiloto) [inline .gitignore:31]: Let us add this after the default `.gitignore`. That means adding this at the very end of the file.

File: doc/source/conf.py [hunk 3]
```diff
@@ -128,3 +138,76 @@
     "body": generate_404(),
 }
 notfound_no_urls_prefix = True
+
+
+# specify the URL of the archive here
+# Url for sphinx-design
+archive_url = "https://github.com/executablebooks/sphinx-design/tree/main/docs/snippets/rst"
```
Comment (jorgepiloto) [inline doc/source/conf.py:145]: This variable is a constant. Hence, it should be declared it in line `209` and wiht upper casing.

File: doc/source/conf.py [hunk 4]
```diff
@@ -128,3 +138,76 @@
...
+        List of example links.
+    """
+    response = requests.get(archive_url)
+    soup = BeautifulSoup(response.content, "html5lib")
+    links = soup.find_all("a")
+    example_links = [
+        "https://raw.githubusercontent.com" + link["href"].replace("/blob/", "/")
+        for link in links
+        if link["href"].endswith(".txt") and not "article-info.txt" in link["href"]
+    ]
```
Comment (jorgepiloto) [inline doc/source/conf.py:169]: Very good usage of list comprehensions. Let me suggests to add a new function parameter named `exclude_files`. It should be a list of strings. Filenames declared in this list should be excluded in line `168`.

File: doc/source/examples/sphinx_examples/.gitignore [hunk 5]
```diff
@@ -0,0 +1,2 @@
+*
```
Comment (jorgepiloto) [inline doc/source/examples/sphinx_examples/.gitignore:1]: Is this file required?

File: pyproject.toml [hunk 6]
```diff
@@ -36,6 +36,11 @@ doc = [
     "Sphinx==5.3.0",
     "sphinx-copybutton==0.5.1",
     "sphinx-notfound-page==0.8.3",
+    "sphinx-design",
```
Comment (jorgepiloto) [inline pyproject.toml:39]: Let us fix the version for `sphinx-design` too. I know this implies downloading the examples from the right release branch, so let us fix that too.

File: doc/source/examples/examples.rst [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+The rendering of sphinx design with ansys sphinx theme. To use this see 
+the full document of `sphinx design <https://sphinx-design.readthedocs.io/en/latest/index.html>`_.
+
+.. jinja:: examples
+
+    {% for filename in inputs_examples %}
+    {% set title = filename.split('.')[0] %}
+
+    {{ title[0].upper() }}{{ title[1:] }}
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (jorgepiloto) [inline doc/source/examples/examples.rst:12]: This line does not seem to adapt to the lenght of the title. Use this trick:

```rst
{{ '~' * (title | length) }}
```

File: doc/source/examples/examples.rst [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+        .. literalinclude:: sphinx_examples/{{ filename }}
+           :language: bash
```
Comment (jorgepiloto) [inline doc/source/examples/examples.rst:15]: Language should be `rst`:

```suggestion
           :language: rst
```

I know that the original examples are in `txt` but their content is `rst`. If we need to modify the file extension when downloading those, let's do that unless it causes problems when building the docs.

**Review by jorgepiloto — CHANGES_REQUESTED** [review]

File: doc/source/examples/sphinx_examples/.gitignore (hunk 5 above)
Comment (Revathyvenugopal162) [reply doc/source/examples/sphinx_examples/.gitignore:1]: I think yes. And we dont need sphinx_design.rst https://github.com/ansys/ansys-sphinx-theme/pull/191#discussion_r1105314511 instead
I was trying to download all the txt files from the sphinx design and download to this directory, so that it will be useful for

1. get the file name from txt name.
2. get the raw directive and rendered docs from the txt files. (useful as a quick reference.)as shown in the SS.
3. Exclude the files if it is showing some error while rendering. 
sphinx-design
But i am open to changes , to keep all the content of text file in single rst.

File: src/ansys_sphinx_theme/theme/ansys_sphinx_theme/static/js/table.js [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+    $('table.datatable').DataTable();
+} );
+
+$(document).ready( function () {
+    $('table.gridtable').GridTable();
+} );
+
+$(document).ready( function () {
+    $('table.gianttables').GiantTable();
+} );
```
Comment (Revathyvenugopal162) [inline src/ansys_sphinx_theme/theme/ansys_sphinx_theme/static/js/table.js:11]:
```suggestion
```

File: pyproject.toml [hunk 10]
```diff
@@ -36,6 +36,11 @@ doc = [
     "Sphinx==5.3.0",
     "sphinx-copybutton==0.5.1",
     "sphinx-notfound-page==0.8.3",
+    "sphinx-design==0.3.0",
+    "bs4==0.0.1",
+    "html5lib",
```
Comment (jorgepiloto) [inline pyproject.toml:41]: May we fix this one too?

File: doc/source/conf.py [hunk 11]
```diff
@@ -128,3 +138,73 @@
...
+
+    Returns
+    -------
+    list
+        List of example links.
+    """
+    response = requests.get(archive_url)
+    soup = BeautifulSoup(response.content, "html5lib")
+    links = soup.find_all("a")
+    exclude_files = ["article-info.txt"]
```
Comment (jorgepiloto) [inline doc/source/conf.py:160]: This must be a parameter in the function.

**Review by jorgepiloto — CHANGES_REQUESTED** [review]
Few minior coments. What about the `particle_background.jpg` file that is commited. Where is that image being used?

**Comment by Revathyvenugopal162** [comment]
> Few minior coments. What about the `particle_background.jpg` file that is commited. Where is that image being used?

That is the background image for sphinx card. in examples

**Review by jorgepiloto — APPROVED** [review]
Awesome implementation, @Revathyvenugopal162. This is going to be really useful for our users. Approving. Feel free to merge 🚀

File: .github/workflows/ci_cd.yml [hunk 12]
```diff
@@ -49,7 +49,7 @@ jobs:
 
   doc-deploy-development:
     name: Doc dev version deploy
-    if: ${{ github.ref == 'refs/heads/main' }}
+    # if: ${{ github.ref == 'refs/heads/main' }}
```
Comment (RobPasMue) [inline .github/workflows/ci_cd.yml:52]:
```suggestion
    if: ${{ github.ref == 'refs/heads/main' }}
```

**Review by RobPasMue — APPROVED** [review]

**Comment by Revathyvenugopal162** [comment]
Thank you @jorgepiloto and @RobPasMue for the thorough review. merging!
